Kitodo-DataFormat/src/main/java/org/kitodo/dataformat/access/DivXmlElementAccess.java

```diff
@@ -199,108 +158,19 @@ private final MdSec amdSecTypeOf(Mets mets, MdSecType mdSec) {
      *            type of meta-data section
      */
     private final void readMetadata(MdSecType mdSec, MdSec mdSecType) {
-        metadata.addAll(mdSec.getMdWrap().getXmlData().getAny().parallelStream().filter(JAXBElement.class::isInstance)
-                .map(JAXBElement.class::cast).map(JAXBElement::getValue).filter(KitodoType.class::isInstance)
-                .map(KitodoType.class::cast)
-                .flatMap(kitodoType -> Stream.concat(
-                    kitodoType.getMetadata().parallelStream()
-                            .map(metadataType -> new MetadataEntry(mdSecType, metadataType)),
-                    kitodoType.getMetadataGroup().parallelStream()
-                            .map(metadataGroupType -> new MetadataEntriesGroup(mdSecType, metadataGroupType))))
-                .collect(Collectors.toList()));
-    }
-
-    /**
-     * Returns the views associated with this structure.
-     * 
-     * @return the views
-     */
-    @Override
-    public List<AreaXmlElementAccessInterface> getAreas() {
-        return views;
-    }
-
-    /**
-     * Returns the substructures associated with this structure.
-     * 
-     * @return the substructures
-     */
-    @Override
-    public List<DivXmlElementAccessInterface> getChildren() {
-        return substructures;
-    }
-
-    /**
-     * Returns the label of this structure.
-     * 
-     * @return the label
-     */
-    @Override
-    public String getLabel() {
-        return label;
-    }
-
-    /**
-     * Returns the meta-data on this structure.
-     * 
-     * @return the meta-data
-     */
-    @Override
-    public Collection<MetadataAccessInterface> getMetadata() {
-        return metadata;
-    }
-
-    /**
-     * Returns the order label of this structure.
-     * 
-     * @return the order label
-     */
-    @Override
-    public String getOrderlabel() {
-        return orderlabel;
-    }
-
-    /**
-     * Returns the type of this structure.
-     * 
-     * @return the type
-     */
-    @Override
-    public String getType() {
-        return type;
-    }
-
-    /**
-     * Sets the label of this structure.
-     * 
-     * @param label
-     *            label to set
-     */
-    @Override
-    public void setLabel(String label) {
-        this.label = label;
-    }
-
-    /**
-     * Sets the order label of this structure.
-     * 
-     * @param orderlabel
-     *            order label to set
-     */
-    @Override
-    public void setOrderlabel(String orderlabel) {
-        this.orderlabel = orderlabel;
-    }
-
-    /**
-     * Sets the type of this structure.
-     * 
-     * @param type
-     *            type to set
-     */
-    @Override
-    public void setType(String type) {
-        this.type = type;
+        super.getMetadata()
+                .addAll(
+                    mdSec.getMdWrap().getXmlData().getAny().parallelStream().filter(JAXBElement.class::isInstance)
```

Comment:
what is happening here?
multiple class casts?
this stream is WAY to long to read and understand it!

Reply:
Thes same as has happened [before](https://github.com/kitodo/kitodo-production/blob/master/Kitodo-DataFormat/src/main/java/org/kitodo/dataformat/access/Structure.java#L202-L210): The metadata is read and written to the data fields. However, this is already existing code and has nothing to do with the current change. If you want to, we can change that, but I would like to do it in a separate pull request.